src/view/actions/updateVariable.jsx

```diff
@@ -60,6 +61,10 @@ const getInitialFormStateFromDataElement = async ({
   existingFormStateNode,
   signal,
 }) => {
+  const value = {};
+  deepAssign(value, context.originalData);
+  deepAssign(value, data);
```

Comment:
Why use `deepAssign` over `Object.assign` or `structuredClone`?

Reply:
Object.assign isn't recursive. Value could be a deeply nested object. I suppose the first deepAssign call could use structuredClone.

Reply:
Since we are using `=` to copy the things from the old object to the new one, the new one is full of references to the old object. So, for example

```js
const original = { a: { b: "alloy" } };
const clone = deepAssign({}, original);
original.a.b = "adobe";
console.log(clone.a.b); // "adobe"
```
Even though we didn't touch the clone.
If we do
```js
const original = { a: { b: "alloy" } };
const clone = structuredClone(original);
original.a.b = "adobe";
console.log(clone.a.b); // "alloy"
```
it works as (you maybe) expected